File: python/2214. Minimum Health to Beat Game/solution.py
class Solution:
    def minimumHealth(self, damage: List[int], armor: int) -> int:
        maxDamage = 0
        health = 1

        for dmg in damage:
            maxDamage = max(maxDamage, dmg)
            health += dmg

        return health - min(maxDamage, armor)

File: java/2214. Minimum Health to Beat Game/Solution.java
class Solution {

    public long minimumHealth(int[] damage, int armor) {

        int maxDamage = Integer.MIN_VALUE;
        long totalDamage = 0;
        for (int dmg : damage) {
            maxDamage = Math.max(maxDamage, dmg);
            totalDamage += dmg;
        }

        return totalDamage - Math.min(armor, maxDamage) + 1;
    }
}